doozerlib/rpmcfg.py

```diff
@@ -140,16 +140,22 @@ def tito_setup(self):
         tito_dist = tito_dist if tito_dist else '.el7aos'
 
         with Dir(self.source_path):
-            if not os.path.isdir(tito_dir):
-                pre_init_sha = exectools.cmd_assert('git rev-parse HEAD')[0].strip()
-                exectools.cmd_assert('tito init')
-                # tito init will try to create a commit. We want this commit to be ignored
-                # from the perspective of any builds about to occur. So, we want to reset
-                # back to the original HEAD. The complexity here is that tito init will NOT
-                # make a commit if .gitignore is configured to somehow ignores the the .tito
-                # directory it creates. Thus, we store the SHA before running tito init
-                # and use that as a means to return to the upstream HEAD.
-                exectools.cmd_assert('git reset {}'.format(pre_init_sha))
+
+            # We either user .tito from source or we don't. If you want to use the upstream
+            # source, use self.config.content.build.use_source_tito_config=True. Otherwise, we
+            # are going to clear it out and create our own.
+            if os.path.isdir(tito_dir):
+                shutil.rmtree(tito_dir)
+
+            pre_init_sha = exectools.cmd_assert('git rev-parse HEAD')[0].strip()
+            exectools.cmd_assert('tito init')
+            # tito init will try to create a commit. We want this commit to be ignored
+            # from the perspective of any builds about to occur. So, we want to reset
+            # back to the original HEAD. The complexity here is that tito init will NOT
+            # make a commit if .gitignore is configured to somehow ignores the the .tito
```

Comment:
```suggestion
            # make a commit if .gitignore is configured to somehow ignore the the .tito
```

Reply:
honestly i would probably reduce this to "clear out anything tito may have added by reverting to previous HEAD"

Reply:
Simplified comment.